addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/handler/AirQualityHandler.java

```diff
@@ -0,0 +1,303 @@
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+
+            // Build a valid URL for the aqicn.org service
+            AirQualityConfiguration config = getConfigAs(AirQualityConfiguration.class);
+
+            String geoStr = location.replace(" ", "").replace(",", ";").trim();
+
+            String urlStr = URL.replace("%APIKEY%", StringUtils.trimToEmpty(config.apikey));
+            urlStr = urlStr.replace("%LOCATION%", geoStr);
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from aqicn.org
+            URL url = new URL(urlStr);
+            URLConnection connection = url.openConnection();
+            String repsonse = IOUtils.toString(connection.getInputStream());
+            logger.debug("aqiData = {}", repsonse);
+
+            // Map the JSON response to an object
+            result = gson.fromJson(repsonse, AirQualityJsonResponse.class);
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getStatus() == "error") {
+                errorMsg = result.getData().toString();
+            } else {
+                resultOk = true;
+
+                if (result.getData() == null) {
+                    resultOk = false;
+                    errorMsg = "missing data sub-object";
+                }
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in aqicn.org (Air Quality) response: {}", errorMsg);
+            }
+
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (IOException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error running aqicn.org (Air Quality) request: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error parsing aqicn.org (Air Quality) response: {}", errorMsg);
+        }
+
+        // Update the thing status
+        if (resultOk && result != null) {
+            String attributions = result.getData().getAttributions();
+            logger.info("Received AirQuality data from aqicn.org. {}", attributions);
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, attributions);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
+        }
+
+        return resultOk ? result : null;
+    }
+
+    public static Object getValue(String channelId, Object data) throws Exception {
+        String[] fields = StringUtils.split(channelId, "#");
+        return getValue(data, fields, 0);
+    }
+
+    /**
+     * Iterates through the fields and returns the getter value.
+     */
+    @SuppressWarnings("all")
+    private static Object getValue(Object data, String[] fields, int index) throws Exception {
```

Comment:
Is reflection really necessary to fetch attributes from the json object? You might run into security manager problems...